### PR TITLE
Added a custom user agent for calls made to sts and iam

### PR DESCRIPTION
**Pull request by PettitWesley**
Basically just copied from the following ECS CLI code:
https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/aws/ecs/client.go#L77
https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/clients/user_agent.go

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: local-container-endpoints/clients/useragent/useragent.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import (
+	"fmt"
+	"runtime"
+
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/awslabs/amazon-ecs-local-container-endpoints/local-container-endpoints/version"
+)
+
+// UserAgentHeader is the header for a User Agent
+const UserAgentHeader = "User-Agent"
```
Comment (efekarakus) [inline local-container-endpoints/clients/useragent/useragent.go]: nit: looks like this can be private since noone else is using it.

**Comment by PettitWesley** [comment]
@efekarakus I don't think there's a way to view the request details to check the header since its made over HTTPS. I'm not super sure about this though.

`make test`  and `make integ` pass.

**Comment by efekarakus** [comment]
@PettitWesley I meant for example can we use the tool locally and see some metric increment in the backend?

File: local-container-endpoints/clients/useragent/useragent.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+// CustomUserAgentHandler returns a http request handler that sets a custom user agent to all aws requests
+func CustomUserAgentHandler() request.NamedHandler {
+	return request.NamedHandler{
+		Name: "ECSCLIUserAgentHandler",
```
Comment (allisaurus) [inline local-container-endpoints/clients/useragent/useragent.go]: C/P leftovers? We should rename this "ECSLocalEndpointsAgentHandler", yes?

Comment (PettitWesley) [reply local-container-endpoints/clients/useragent/useragent.go]: ooops

File: go.sum [hunk 3]
```diff
@@ -1,6 +1,7 @@
 github.com/Microsoft/go-winio v0.4.12/go.mod h1:VhR8bwka0BXejwEJY73c50VrPtXAaKcyvVC4A4RozmA=
 github.com/aws/amazon-ecs-agent v1.26.0 h1:kT/JZYXpCUXSjNaTWrYjlCPayWCV6dXaMQKOYq5JLfo=
 github.com/aws/amazon-ecs-agent v1.26.0/go.mod h1:as/yyU3OZwCtKXO0Bv9dieYPupNo7jnNNZRNQTu9Row=
+github.com/aws/amazon-ecs-cli v1.13.0 h1:cOOwtpChp2JhDK7PkIoHN5OKNhCPSQhaL7n3Pz7nIYk=
```
Comment (allisaurus) [inline go.sum]: can this be removed? I don't see anywhere new that's using it

Comment (PettitWesley) [reply go.sum]: huh.... this is autogenerated.. not sure why it didn't get auto-removed when I re-built after removing it from go.mod

Comment (PettitWesley) [reply go.sum]: fixed

**Pull request merged**